midend/interpreter.cpp

```diff
@@ -730,6 +730,18 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         clone->expr = li->constant;
         DoConstantFolding cf(refMap, typeMap);
         cf.setCalledBy(this);
+        if (auto cast = expression->to<IR::Cast>()) {
+            auto bitsType = cast->destType->to<IR::Type_Bits>();
+            const IR::Constant* constant;
+            auto resConst = clone->expr->to<IR::Constant>();
+            if (resConst) {
+                constant = new IR::Constant(bitsType, resConst->value);
```

Comment:
why are you doing this?
The call to DoConstantFolding above should be doing exactly the same thing below, when it calls apply(cf).